workspaces/src/network/testnet.rs

```diff
@@ -67,15 +68,25 @@ impl std::fmt::Debug for Testnet {
 impl AllowDevAccountCreation for Testnet {}
 
 #[async_trait]
-impl TopLevelAccountCreator for Testnet {
-    async fn create_tla(
+impl DevAccountCreator for Testnet {
+    async fn create_dev_account(
         &self,
         worker: Worker<dyn Network>,
         id: AccountId,
         sk: SecretKey,
         // TODO: return Account only, but then you don't get metadata info for it...
     ) -> Result<Execution<Account>> {
         let url = Url::parse(HELPER_URL).unwrap();
+        //only registrar can create tla on testnet, so must concatenate random created id with .testnet
+        let id = if self.info().name.eq("testnet") {
+            AccountId::from_str(format!("{}.testnet", id.as_str()).as_str())
```

Comment:
I suggest to avoid unnecessary hardcodings:

```suggestion
        let id = AccountId::from_str(format!("{}.{}", id, self.into().root_id)).map_err(...)?;
```